extensions/amp-ad-exit/0.1/amp-ad-exit.js

```diff
@@ -90,28 +93,72 @@ export class AmpAdExit extends AMP.BaseElement {
    * @return {function(string): string}
    */
   getUrlVariableRewriter_(args, event, target) {
-    const vars = {
+    const substitutionFunctions = {
       'CLICK_X': () => event.clientX,
       'CLICK_Y': () => event.clientY,
     };
+    const replacements = Services.urlReplacementsForDoc(this.getAmpDoc());
     const whitelist = {
       'RANDOM': true,
       'CLICK_X': true,
       'CLICK_Y': true,
     };
     if (target.vars) {
-      for (const customVar in target.vars) {
-        if (customVar[0] == '_') {
-          vars[customVar] = () =>
-              args[customVar] == undefined ?
-                target.vars[customVar].defaultValue : args[customVar];
-          whitelist[customVar] = true;
+      for (const customVarName in target.vars) {
+        if (customVarName[0] == '_') {
+          const customVar =
+              /** @type {!./config.Variable} */ (target.vars[customVarName]);
+          if (customVar) {
+            /*
+             Example:
+             The amp-ad-exit target has a variable representing the
+             priority of something, which is defined as follows:
+             "vars": {
+             "_pty": {
+             "defaultValue": "unknown",
+             "iframeTransportSignal":
+             "IFRAME_TRANSPORT_SIGNAL(vendorXYZ,priority)"
+             },
+             ...
+             }
+             The cross-domain iframe of vendorXYZ has sent the
+             following response for the creative:
+             { priority: medium, category: W }
+             This is just example data. The keys/values in that object can
+             be any strings.
+             The code below will create substitutionFunctions['_pty'],
+             which in this example will return "medium".
+            */
+            substitutionFunctions[customVarName] = () => {
+              if ('iframeTransportSignal' in customVar) {
```

Comment:
Same here... consider just returning if not in

Reply:
Discussed verbally. Can invert, but would be wrong to just return if condition is false due to args and default value.